### ISSUE TITLE
Implement train splitting / joining as GTFS transfers

### ISSUE THREAD

**Issue by miklcct** [planarnetwork/dtd2mysql#81]
https://github.com/google/transit/pull/303 adds the ability to specify [same-vehicle transfer](https://gtfs.org/schedule/reference/#linked-trips) in GTFS. Splitting and joining trains within a passenger service can be specified with `transfer_type` = `4` for the constituent portions.

**Comment by linusnorton** [comment]
Okay, so rather than have the new services, just populate transfers.txt with same vehicle transfers? Seems like a good idea to me.

**Comment by miklcct** [comment]
For a service A-B-C splitting off a B-D portion, the specification models it as a A-B service, B-C service and B-D service, which the A-B service joins with both B-C and B-D service.

**Comment by joshuajaharwood** [comment]
Heya @linusnorton, would you like me to implement this? I'm looking to make more improvements to the project but I understand they need review so I'd like to start with the most important stuff first :)

**Comment by linusnorton** [comment]
Yes, that would be good. I'll make other tickets for it but I'd also like to do two things:

1) Make a direct dtd to gtfs translation (I believe at the moment it needs a db). 
2) Add support for postgres and other databases

At which point, I think we'd need to rename the project. Maybe dtd-feed-processor or something.

**Comment by miklcct** [comment]
> Yes, that would be good. I'll make other tickets for it but I'd also like to do two things:
> 
> 1. Make a direct dtd to gtfs translation (I believe at the moment it needs a db).
> 2. Add support for postgres and other databases
> 
> At which point, I think we'd need to rename the project. Maybe dtd-feed-processor or something.

In such case I would like to be a collaborator. During my time at Aubin I made a fork Aubin-MaaS/dtd2mysql which changed the GTFS generation to be compatible with GTFS feeds for other modes of transport, using pseudo-ATCO codes for stop_id and also correcting the train number, headsign and platform number handlings as well. It is currently used in Aubin and Transitous, along with internal real-time data, to provide GTFS data for National Rail identical to the official information and allows for multi-modal integration at places such as Amersham and Wimbledon.

**Comment by joshuajaharwood** [comment]
That functionality would be very useful for my use cases also. I'll take the multi DB support if you aren't opposed?

**Comment by miklcct** [comment]
> That functionality would be very useful for my use cases also. I'll take the multi DB support if you aren't opposed?

That will be good.

**Comment by linusnorton** [comment]
@miklcct should we revisit your previous PR? My only real concern with it was changing the stop codes, I know people that rely on the CRS code being in a particular field. Was that part of the change required?

**Comment by miklcct** [comment]
> [@miklcct](https://github.com/miklcct) should we revisit your previous PR? My only real concern with it was changing the stop codes, I know people that rely on the CRS code being in a particular field. Was that part of the change required?

The fork as further developed from that PR so I think it is better to work from that repository.

In Aubin, the CRS code is in the GTFS `stop_code` field, which is the expected behaviour of a passenger searching for timetables. However, in the original repository, the Tiploc code is placed there instead.

This change is required from the user's perspective according to the GTFS standard, as CRS codes are used by end users to identity stops.

**Comment by linusnorton** [comment]
Okay, I can speak to some people and then we can put it down as a major version bump.
